### PR TITLE
Test Windows in CI

### PR DESCRIPTION
**Pull request by trask**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -68,7 +69,8 @@ jobs:
             ./gradlew build
             ${{ matrix.coverage && 'jacocoTestReport' || '' }}
             -PtestJavaVersion=${{ matrix.test-java-version }}
-            -Porg.gradle.java.installations.paths=${{ steps.setup-java-test.outputs.path }},${{ steps.setup-java.outputs.path }}
+            "-Porg.gradle.java.installations.paths=${{ steps.setup-java-test.outputs.path }}"
+            "-Porg.gradle.java.installations.auto-download=false"
```
Comment (trask) [inline .github/workflows/build.yml:73]: I think the double quotes ended up being the only thing that was needed here, but the other changes seem good and align with the opentelemetry-java-instrumentation project (where, btw, we only run smoke tests against windows but not regular integration or unit tests, so haven't run into this double quote issue)

File: context/src/test/java/io/opentelemetry/context/internal/shaded/WeakConcurrentMapTest.java [hunk 2]
```diff
@@ -152,7 +152,7 @@ void doTest() throws Exception {
       assertThat(values.isEmpty(), is(true));
       key1 = key2 = null; // Make eligible for GC
       System.gc();
-      Thread.sleep(200L);
+      Thread.sleep(1000L);
```
Comment (jkwatson) [inline context/src/test/java/io/opentelemetry/context/internal/shaded/WeakConcurrentMapTest.java:155]: could we also change the asserts below to be under an `await()` ?

Comment (trask) [reply context/src/test/java/io/opentelemetry/context/internal/shaded/WeakConcurrentMapTest.java:155]: I _think_ this sleep is more about waiting for the gc to complete before running `triggerClean()` below

Comment (trask) [reply context/src/test/java/io/opentelemetry/context/internal/shaded/WeakConcurrentMapTest.java:155]: alternatively, I could add the `triggerClean()` into the `await()` block, I'll try that

**Review by jkwatson — APPROVED** [review]

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -68,15 +69,16 @@ jobs:
...
         env:
           GRADLE_ENTERPRISE_ACCESS_KEY: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}
           # JMH-based tests run only  if this environment variable is set to true
           RUN_JMH_BASED_TESTS: ${{ matrix.jmh-based-tests }}
 
       - name: Check for diff
         # The jApiCmp diff compares current to latest, which isn't appropriate for release branches, or for bot-generated PRs
-        if: ${{ !startsWith(github.ref_name, 'release/') && !startsWith(github.base_ref, 'release/') && (github.actor != 'opentelemetrybot') }}
+        if: ${{ matrix.os != 'windows-latest' && !startsWith(github.ref_name, 'release/') && !startsWith(github.base_ref, 'release/') && (github.actor != 'opentelemetrybot') }}
```
Comment (jack-berg) [inline .github/workflows/build.yml:81]: #nit: maybe update the comment to specify why its not required on windows?

File: sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/AutoConfiguredOpenTelemetrySdkBuilder.java [hunk 4]
```diff
@@ -541,13 +541,7 @@ private static AutoConfiguredOpenTelemetrySdk maybeConfigureFromFile(
...
     }
     logger.fine("Autoconfiguring from configuration file: " + configurationFile);
-    FileInputStream fis;
-    try {
-      fis = new FileInputStream(configurationFile);
-    } catch (FileNotFoundException e) {
-      throw new ConfigurationException("Configuration file not found", e);
-    }
-    try {
+    try (FileInputStream fis = new FileInputStream(configurationFile)) {
```
Comment (jack-berg) [inline sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/AutoConfiguredOpenTelemetrySdkBuilder.java:544]: Nice catch here to close this. Is is really the case that `IOException` is only thrown when an error occurs closing the file?

**Review by jack-berg — APPROVED** [review]

File: .github/workflows/build.yml (hunk 3 above)
Comment (trask) [reply .github/workflows/build.yml:86]: added comment

File: sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/AutoConfiguredOpenTelemetrySdkBuilder.java (hunk 4 above)
Comment (trask) [reply sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/AutoConfiguredOpenTelemetrySdkBuilder.java:544]: yeah, I was kinda surprised by that too, I'll add a comment in that catch block

**Pull request merged**